### ISSUE TITLE
Bug: [consistent-type-imports] False positive when using DI and decorators (tsyringe)

### ISSUE THREAD

**Issue by ajmnz** [typescript-eslint/typescript-eslint#5903]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=GYVwdgxgLglg9mABDMArAptAhgIwDboA8AKgHwAUAlAFyLlRYBOA5ulLcZYgLymIBucGABNEAbwBQiRIzYhGSUJFgI6DFmw40BQ0ZOnSpiAL4BuCcYkSA9NcQABKAGcAtDGZg4siTAC2ABy8ocUQAISYTRGBGOF9EAHIAOmscJnjzCXsUDGx8dCoJCDwsJydEADE4OHEjCAQnKEYQaC9yf0YYfiwodERUxlpwxi4xS2MgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyo0kRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjp4yPABF+sTAEM8KgLLo8qgCbrVFMKIlTa6AB6IsyeOiK7USuGo2ZhJzGKkBfEL5AA

### Repro Code

```typescript
function injectable<T>(): (target: T) => void {
  return function (target: T): void {
    
  };
}

import { Bar } from './bar'; // Error: All imports in the declaration are only used as types.

@injectable()
class Foo {
  constructor(private bar: Bar) {}
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-imports": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  }
}
```

### Expected Result

I expected that #2751 (#2559) would've fixed this and the rule wouldn't have assumed that it is only used as a type, but turns out it only catches decorators from within the constructor:

```ts
import { Bar } from './bar'; // ok

class Foo {
  constructor(@deco(() => Bar) private bar: Bar) {}
}
```

```ts
import { Bar } from './bar'; // Expected ok, but got Error: All imports in the declaration are only used as types.

@injectable()
class Foo {
  constructor(private bar: Bar) {}
}
```

### Actual Result

The rule assumes `Bar` is used as a type, causing the DI library (in the case [tsyringe](https://github.com/Microsoft/tsyringe)) to be unable to inject the dependencies (`Bar`) at runtime. This is pretty similar to the issue with Nest.js (#2559)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.41.0` |
| `@typescript-eslint/parser`            | `playground default` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `playground default` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/README.md#parseroptionsemitdecoratormetadata

You need to turn this on in the parser options if you're not using type-aware linting.
